core/trino-main/src/main/java/io/trino/operator/SimplePagesHashStrategy.java

```diff
@@ -60,7 +61,7 @@ public SimplePagesHashStrategy(
     {
         this.types = ImmutableList.copyOf(requireNonNull(types, "types is null"));
         this.comparisonOperators = types.stream()
-                .map(blockTypeOperators::getComparisonOperator)
+                .map(type -> type.isOrderable() ? Optional.of(blockTypeOperators.getComparisonOperator(type)) : Optional.<BlockPositionComparison>empty())
```

Comment:
is `<BlockPositionComparison>` needed here?

Reply:
Yes, it's needed.
```
[ERROR] /Users/takezoe/trino/core/trino-main/src/main/java/io/trino/operator/SimplePagesHashStrategy.java:[65,25] incompatible types: inference variable E has incompatible bounds
[ERROR]     equality constraints: java.util.Optional<io.trino.type.BlockTypeOperators.BlockPositionComparison>
[ERROR]     lower bounds: java.util.Optional<? extends java.lang.Object>
```